packages/dd-trace/src/config.js

```diff
@@ -806,31 +544,289 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
+    this._setBoolean(env, 'telemetry.logCollection', coalesce(DD_TELEMETRY_LOG_COLLECTION_ENABLED, DD_IAST_ENABLED))
+    this._setBoolean(env, 'telemetry.metrics', DD_TELEMETRY_METRICS_ENABLED)
+    this._setBoolean(env, 'traceId128BitGenerationEnabled', DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED)
+    this._setBoolean(env, 'traceId128BitLoggingEnabled', DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED)
     this._setBoolean(env, 'tracing', DD_TRACING_ENABLED)
+    this._setString(env, 'version', DD_VERSION || tags.version)
   }
 
   _applyOptions (options) {
     const opts = this._options = this._options || {}
     const tags = {}
 
-    options = Object.assign({ ingestion: {} }, options, opts)
+    options = this.options = Object.assign({ ingestion: {} }, options, opts)
 
     tagger.add(tags, options.tags)
 
-    this._setString(opts, 'service', options.service || tags.service)
+    this._setValue(opts, 'appsec.blockedTemplateHtml', maybeFile(options.appsec.blockedTemplateHtml))
+    this._setValue(opts, 'appsec.blockedTemplateJson', maybeFile(options.appsec.blockedTemplateJson))
+    if (options.appsec.rules) this._setBoolean(opts, 'appsec.customRulesProvided', true)
+    this._setBoolean(opts, 'appsec.enabled', options.appsec.enabled)
+    this._setString(opts, 'appsec.obfuscatorKeyRegex', options.appsec.obfuscatorKeyRegex)
+    this._setString(opts, 'appsec.obfuscatorValueRegex', options.appsec.obfuscatorValueRegex)
+    this._setValue(opts, 'appsec.rateLimit', maybeInt(options.appsec.rateLimit))
+    this._setString(opts, 'appsec.rules', options.appsec.rules)
+    this._setValue(opts, 'appsec.wafTimeout', maybeInt(options.appsec.wafTimeout))
+    this._setBoolean(opts, 'clientIpEnabled', options.clientIpEnabled)
+    this._setString(opts, 'clientIpHeader', options.clientIpHeader)
+    this._setString(opts, 'dbmPropagationMode', options.dbmPropagationMode)
+    if (options.dogstatsd) {
+      this._setString(opts, 'dogstatsd.hostname', options.dogstatsd.hostname)
+      this._setString(opts, 'dogstatsd.port', options.dogstatsd.port)
+    }
+    this._setBoolean(opts, 'dsmEnabled', options.dsmEnabled)
     this._setString(opts, 'env', options.env || tags.env)
-    this._setString(opts, 'version', options.version || tags.version)
-    this._setUnit(opts, 'sampleRate', coalesce(options.sampleRate, options.ingestion.sampleRate))
-    this._setBoolean(opts, 'logInjection', options.logInjection)
+    this._setBoolean(opts, 'experimental.enableGetRumData',
+      options.experimental && options.experimental.enableGetRumData)
+    this._setString(opts, 'experimental.exporter', options.experimental && options.experimental.exporter)
+    this._setBoolean(opts, 'experimental.runtimeId', options.experimental && options.experimental.runtimeId)
+    if (!isNaN(parseInt(options.flushInterval, 10))) {
+      this._setValue(opts, 'flushInterval', parseInt(options.flushInterval, 10))
+    }
```

Comment:
why not use `maybeInt()` ?

```suggestion
    this._setValue(opts, 'flushInterval', maybeInt(options.flushInterval))
```

Reply:
can we just take out the radix (`10`)?

Reply:
It's 10 by default, but some people recommend putting it just for clarity, up to you